vignettes/advanced.Rmd

```diff
@@ -26,102 +29,290 @@ knitr::opts_chunk$set(
+
+The simulated trial explained in section \@ref(sec:dataSimul) assumes that outcomes in the active arm observed after the ICE "treatment discontinuation" follow the increments observed in the control arm.
+Thus the imputation of missing data in the active arm after treatment discontinuation might be performed under the CIR assumption. How `rbmi` deals with observed values after an ICE handled using a reference-based method, is to include them in the analysis model but exclude them from the imputation model.
+
+Let's first set-up the estimator: Suppose that imputation model has the mean change from baseline as the outcome, includes the treatment group, the (categorical) visit, treatment-by-visit interactions, the baseline outcome, and baseline outcome-by-visit interactions as covariates, and assumes a common unstructured covariance matrix in both groups.
+The analysis model is ANCOVA that also adjusts for the baseline outcome value. Below we report the code to set the variables of the imputation and analysis models.
+For details on the basic set-up to call the key rbmi functions, please refer to the "quickstart" vignette (`vignette("quickstart")`).
+
+```{r}
+
+# Create data_ice: CIR strategy for each patient in the active arm who stops the treatment
+data_ice_CIR <- data %>%
+    group_by(id = factor(id, levels = unique(id))) %>%
+    mutate(
+      is_interv = group == "Intervention",
+      has_ice = any(ind_ice1 == 1)
+    ) %>%
+    filter(is_interv & has_ice) %>% # Select pts in active arm that disc from treatment
+    summarise(
+        visit = visit[ind_ice1 == 1][1] # Select first visit affected by the ICE
+    )
+data_ice_CIR$strategy <- "CIR"
+
+# Compute change from baseline
+data$change <- data$outcome - data$outcome_bl
+
+# Remove rows corresponding to baseline visit
+data <- data[data$visit %in% levels(data$visit)[-1], ]
+data$visit <- factor(data$visit, levels <- unique(data$visit))
+
 vars <- set_vars(
-    outcome = "outcome",
-    visit = "visit",
     subjid = "id",
+    visit = "visit",
+    outcome = "change",
     group = "group",
-    covariates = c("age", "sex", "group*visit")
+    covariates = c("visit*outcome_bl", "visit*group"),
+    strategy = "strategy"
+    )
+
+vars_an <- vars
+vars_an$covariates <- "outcome_bl"
+
+```
+
+Let's assume that the chosen multiple imputation (MI) method is the approximate Bayesian MI with 10 imputations that can be set with the function `method_approxbayes()` as follows:
+
+```{r}
+method <- method_approxbayes(n_sample = 10)
+```
+
+We can now call sequentially the 4 key functions of `rbmi` to perform the multiple imputation.
+Please note that the management of post-discontinuation observed data is performed without additional complexity for the user.
+`draws()` automatically excludes post-ICE data handled with a reference-based method using information provided by the argument `data_ice`.
```

Comment:
> with a reference-based method

Was just thinking that we only keep "MAR" data. But we still exclude LMCF data but this isn't strictly a reference method is it ?

Reply:
I think it is correct to exclude post-ICE data that follow a LMCF assumption, since this is not compatible with a MAR assumptions assumed by the base imputation model (MMRM)